contracts/beyondFramework/dca/portfolios/PercentageAllocationPortfolio.sol

```diff
@@ -0,0 +1,323 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { IDCAPortfolio } from "../interfaces/IDCAPortfolio.sol";
+import { IDCAInvestable } from "../interfaces/IDCAInvestable.sol";
+import { PortfolioAccessBaseUpgradeable } from "../base/PortfolioAccessBaseUpgradeable.sol";
+
+import { IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/interfaces/IERC20Upgradeable.sol";
+import { SafeERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import { PausableUpgradeable } from "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+import { ReentrancyGuardUpgradeable } from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+contract PercentageAllocationPortfolio is
+    PortfolioAccessBaseUpgradeable,
```

Comment:
I remember solidity docs says the most general classes needs to be put first otherwise the S3 linearization can fail. Let me give you the details in public.